add missing dep

claws opened the pull request:
To actually use the plugin provided by this library the **grpcio-tool** package needs to be installed. Therefore, it should be listed in the **install_requires** field to ensure this library is usable once installed.

vmagamedov commented:
Thank you for your contribution! I think that it would be wrong to require people to install `grpcio-tools` along with `grpcio` to use this library (in runtime). So other projects, which will depend on `grpclib`, will also be forced to depend on `grpcio-tools` and `grpcio` transitively. Maybe `extras_require` setting will be better?

vmagamedov commented:
But the root cause of this issue is probably a lack of documentation. For example, `grpcio-tools` is not the only way to generate code. There is also a `protoc` executable, which comes with `protobuf` system package:
```
pip install grpcio-tools
python -m grpc_tools.protoc ...
```
is the same as:
```
brew install protobuf
protoc ...
```

claws commented:
Yeah, that is likely the case, I was trying to follow the instructions in the README.rst to recreate the sequence of steps you used to generate the files in the *example* directory that are produced by protoc. The line clearly uses a Python module called *grpc_tools* which made me think that it was a missing requirement. I wanted to follow your work to produce the files to understand the workflow for when I use this library on my own *.proto* files. 

I performed the following:

``` console
$ git clone https://github.com/vmagamedov/grpclib.git
$ cd grpclib
$ python3.6 -m venv ../grpclibenv
$ source ../grpclibenv/bin/activate
(grpclibenv) $ pip install -e .
```

Then I tried to follow the instructions for how to produce the files generated by protoc, which failed. 
``` console
(grpclibenv) $ cd example
(grpclibenv) $ python -m grpc_tools.protoc -I. --python_out=. --python_grpc_out=. helloworld.proto
/path/to/venvs/grpclibenv/bin/python: Error while finding module specification for 'grpc_tools.protoc' (ModuleNotFoundError: No module named 'grpc_tools')
```

So, without knowing a lot about protoc I tried to install *grpcio* hoping that it would get me *grpc_tools* module needed by the instructions. But this was incorrect. I then used Google to find that I needed *grpcio-tools* to get the *grpc_tools* module, which allowed the command to run without error.

``` console
(grpclibenv) $ pip install grpcio-tools
(grpclibenv) $ python -m grpc_tools.protoc -I. --python_out=. --python_grpc_out=. helloworld.proto
```

I notice that the files produced indicate that there are differences to the *.protoc* generated files you have checked in. Would you expect a difference? Why have you checked in files generated by the protoc compiler?

``` console
(grpclibenv) $ git status
...
	modified:   helloworld_grpc.py
	modified:   helloworld_pb2.py
...
```

Maybe you could add instructions for building the examples such as:
``` console
# The protoc command line tool is required. There are multiple methods available
# to install protoc. One method, familiar to Python developers, is to use pip. 
$ pip install grpcio-tools
$ cd examples
$ python -m grpc_tools.protoc -I. --python_out=. --python_grpc_out=. helloworld.proto
```

It seems reasonable to me that Python developers using this library would use the Python package approach to call protoc to generate the files.

vmagamedov commented:
Checked-in files were generated by `grpcio-tools==1.4.0`, after upgrading to `grpcio-tools==1.6.0` I'm also seeing the difference: `*_pb2.py` now contains also service definitions.

vmagamedov commented:
Added installation instructions to the README, can you check them? Am I missed something?

claws commented:
Its getting clearer. I guess from my perspective, as a user, I am wanting to follow some steps to run the example contained in the project so I can understand the workflow required to use this library.

A typical user of this library will need to generate stubs for their own project (which reinforces my expectation that this library should include the grpcio-tools package in the *install_requires* field) so I was kind of expecting the example to walk through the process. That would certainly help me in this case.

The command line example in the README used to inform the reader about how to produce the generated output seems incorrect - with regard to producing output identical to the protoc generated code you have checked in. 

If I run it as written, I end up with the imports looking different which makes me think that the instruction is inaccurate or is perhaps missing a precursor step. If I run a slightly modified version of the command:

``` console
$ python -m grpc_tools.protoc -I. --python_out=. --python_grpc_out=. example/helloworld.proto
```

I get output looking closer to the generated files you have checked in. Notice in my command above how I have run this command from the top level directory and had to include the *example* directory before the helloworld.proto? In the existing README it seems to imply that the command should be run in the *example* directory - but then the imports are different.

It might be fine either way perhaps, I haven't really worked out how to run the example to test it. I get the following errors when I try to run the example client and server:

``` console
$ cd example
$ python server.py 
Traceback (most recent call last):
  File "server.py", line 5, in <module>
    from .helloworld_pb2 import HelloReply
ModuleNotFoundError: No module named '__main__.helloworld_pb2'; '__main__' is not a package
```
and
``` console
$ cd example
$ python client.py 
Traceback (most recent call last):
  File "client.py", line 8, in <module>
    from .helloworld_pb2 import HelloRequest
ModuleNotFoundError: No module named '__main__.helloworld_pb2'; '__main__' is not a package
```

vmagamedov commented:
Seems like it was not clear that ``example`` is not just a directory, but a package. I'm refactored example, so now it contains it's own README and ``greeter`` service package. README contains instructions about how to generate code in the ``example`` directory (now just directory) and how to run example. And from now I will assume that examples should be usable inside ``example`` directory, and not from the root directory (because earlier ``example`` was also a package).

vmagamedov commented:
Closing this PR because `grpcio-tools` is not required by `grpclib` in runtime.
